pkg/config/expand.go

```diff
@@ -573,11 +573,11 @@ func (ref *modReference) validate(depGroups []DeploymentGroup, modToGrp map[stri
 // ref.ExplicitInterGroup: intergroup references must explicitly identify the
 // target group ID and intragroup references cannot have an incorrect explicit
 // group ID
-func validateReference(ref varReference, context varContext, modToGrp map[string]int) error {
+func (ref *varReference) validate(depGroups []DeploymentGroup, vars map[string]interface{}, modToGrp map[string]int) error {
```

Comment:
nit: This looks similar enough to me that I feel like there is an opportunity for code reuse between this and modReference. Something like an interface that does `validateModuleReference` and then the varReference would do additional validation. Probably not congruent with my other comment about `ID` name.

I think it would look something like [this](https://stackoverflow.com/q/31997753) in practice.

I realize I already brought this up before and we moved past it. So don't feel like you need to do it.

Reply:
I do want to consider a `type reference interface {...}` as a means of implementing `validate`. At the moment, that is complicated by the fact that the parameter list is different for each. I prefer to tackle that as its own task, probably as part of a general attempt to simplify blueprint parsing and validation.